### ISSUE TITLE
Provide more helpful error message during archive-stop situations

### ISSUE THREAD

**Issue by terrorobe** [pgmasters/backrest#83]
Currently, when in archive-stop conditions the DBA doesn't get any feedback on why it entered this state: 

```
postgres@prd-pgbackslave-02-tor ~ $ /opt/backrest/bin/pg_backrest.pl --stanza=backslave-02 archive-push pg_xlog/000000010000000000000071
2015-05-27 09:12:53.669 T00  ERROR: archive stop file (/data/postgres_backups/lock/backslave-02-archive.stop) exists , discarding 000000010000000000000071
2015-05-27 09:12:53.669 T00  ERROR: process terminated on signal or exception, 0 threads stopped
Undefined subroutine &BackRest::Archive::remote_exit called at /opt/backrest/bin/../lib/BackRest/Archive.pm line 360.
 at /opt/backrest/bin/pg_backrest.pl line 285.
```

Only after removing the lock file you get a reason:

```
postgres@prd-pgbackslave-02-tor ~ # rm /data/postgres_backups/lock/backslave-02-archive.stop
postgres@prd-pgbackslave-02-tor ~ $ /opt/backrest/bin/pg_backrest.pl --stanza=backslave-02 archive-push pg_xlog/000000010000000000000071
2015-05-27 09:13:11.875 T00   INFO: pushing WAL segment pg_xlog/000000010000000000000071 asynchronously
2015-05-27 09:13:12.014 T00   INFO: starting async archive-push
postgres@prd-pgbackslave-02-tor ~ $ 2015-05-27 09:13:12.679 T00  ERROR: local archive store has exceeded limit of 128MB, archive logs will be discarded
2015-05-27 09:13:12.679 T00   INFO: archive to be copied to backup total 14, size 224MB
2015-05-27 09:13:12.681 T00  ERROR: [120] : WAL segment 000000010000000000000064 already exists in the archive
postgres@prd-pgbackslave-02-tor ~ $
```

If ``archive-max-mb`` is currently the only cause for archive-stop maybe just add a reference to the "archive stop file exists" message

**Comment by dwsteele** [comment]
I think this is related to issue #84.  If I fix that this should be fixed as well.

**Comment by dwsteele** [comment]
Fixed in 3083e5f.
